src/app-host/live-reload-client.js

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Based in part on code from Vogue (https://github.com/andrewdavey/vogue)
+
+module.exports.start = function (sock) {
+    var hop = Object.prototype.hasOwnProperty;
+    var head = document.getElementsByTagName('head')[0];
+    var serverUrl = window.location.protocol + '//' + window.location.host + '/';
+    var stylesheets;
+    var socket = sock;
+
+    /**
+     * Reload a stylesheet. We just add a new query parameter to the stylesheet's URL, which causes the browser to fetch it again and apply it.
+     *
+     * @param {String} sheetRelativePath The URL of the stylesheet to be reloaded.
+     */
+    function reloadStylesheet(sheetRelativePath) {
+        if (!stylesheets[sheetRelativePath]) {
+            return;
+        }
+
+        var newHref = sheetRelativePath + (sheetRelativePath.indexOf('?') >= 0 ? '&' : '?') + '_livereload=' + (new Date).getTime();
+
+        // Check if the appropriate DOM Node is there.
+        if (!stylesheets[sheetRelativePath].setAttribute) {
+            // Create the link.
+            var stylesheet = document.createElement('link');
+
+            stylesheet.setAttribute('rel', 'stylesheet');
+            stylesheet.setAttribute('href', newHref);
+            head.appendChild(stylesheet);
+
+            // Update the reference to the newly created link.
+            stylesheets[sheetRelativePath] = stylesheet;
+        } else {
+            // Update the href to the new URL.
+            stylesheets[sheetRelativePath].href = newHref;
+        }
+    }
+
+    /**
+     * Reload the page. Currently, only does a naive window.location.reload().
+     */
+    function reloadPage() {
+        window.location.reload(true);
+    }
+
+    /**
+     * Fetch all the local stylesheets from the page.
+     *
+     * @returns {Object} The list of local stylesheets keyed by their relative path.
+     */
+    function getLocalStylesheets() {
```

Comment:
What is the livereload behavior for internal (`<style>` tag in HTML `<head>` tag) and inline CSS?